### PR TITLE
[gpctl] Add debug command

### LINKED ISSUE #6479: Add `gpctl debug <component>` which enables the debug log level for a component
For most components that would be equivalent to
```bash
kubectl port-forward <pod> 6060
curl -XPOST -d '{"level":"debug"}' localhost:6060/debug/logging
```

For supervisor we'd modify the default workspacetemplate and set `SUPERVISOR_DEBUG` to `true`.

### PR DESCRIPTION
**Pull request by csweichel**
## Description
Adds `gpctl debug log <component>` which enables the debug logging for that component.

## Related Issue(s)

Fixes #6479

## How to test
```
gpctl debug log image-builder-mk3
```
start an image build

## Release Notes

```release-note
[gpctl] Add debug command to aid in debugging components
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **csweichel** after the PR has been reviewed.

Associated issue: *#6479*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[dev/gpctl/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/dev/gpctl/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by MrSimonEmms — APPROVED** [review]
/lgtm

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: ac37f115dbc244135d4e1ff5954b238efa605e06</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6480#pullrequestreview-794334556" title="LGTM">MrSimonEmms</a>*

Associated issue: *#6479*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[dev/gpctl/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/dev/gpctl/OWNERS)~~ [MrSimonEmms]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6480#issuecomment-956329762" title="Approved">MrSimonEmms</a>*

Associated issue: *#6479*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[dev/gpctl/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/dev/gpctl/OWNERS)~~ [MrSimonEmms]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
